demos/glacial_isostatic_adjustment/gia_adjoint_ice/gia_adjoint_ice.py

```diff
@@ -0,0 +1,717 @@
+# Synthetic ice inversion using adjoints
+# =======================================================
+#
+# In this tutorial, we will use G-ADOPT's adjoint capability to invert for a synthetic ice load in an annulus domain. We will be running a 'twin' experiment where we will try to recover the ice load that we used as part of the earlier 2d cylindrical tutorial, starting from a different initial guess of the ice load.
+#
+# This example focusses on setting up an adjoint problem. These can be summarised as follows:
+# 1. Defining an objective function.
+# 2. Verifying the accuracy of the gradients using a Taylor test.
+# 3. Setting up and solving a gradient-based minimisation problem for a synthetic ice load.
+
+# This example
+# -------------
+# Let's get started!
+# The first step is to import the gadopt module, which
+# provides access to Firedrake and associated functionality.
+
+from gadopt import *
+from gadopt.utility import step_func, vertical_component, CombinedSurfaceMeasure
+import pyvista as pv
+import matplotlib.pyplot as plt
+
+
+# To bring in G-ADOPT's adjoint functionality we need to start taping the forward problem, which we do below. It's also good practice to clear the tape, so that we are starting fresh each time.
+
+from gadopt.inverse import *
+tape = get_working_tape()
+tape.clear_tape()
+
+# In this tutorial we are going load the mesh created by the forward cylindrical demo in the previous tutorial. This makes it easier to load the synthetic data from the previous tutorial for our 'twin' experiment.
+
+# Set up geometry:
+checkpoint_file = "./forward-2d-cylindrical-disp-incdisp-dt1ka.h5"
```

Comment:
i.e. the `glacial_isostatic_adjustment/2d_cylindrical`? It would be clearer if this filename matched what that one outputs. Then we can make sure all the dependency ordering is in place (see the other adjoint demo). And then we can also pop the checkpoint on gadopt.org.

Reply:
So it's not quite the same output! I tried to speed up the adjoint demo by taking fewer timesteps (dt = 1000 years not 250 years) so the forward run and henche the checkpoint is a bit different